UI/auth-youtube.cpp

```diff
@@ -0,0 +1,223 @@
+#include "auth-youtube.hpp"
+
+#include <iostream>
+#include <QMessageBox>
+#include <QThread>
+#include <vector>
+#include <QHBoxLayout>
```

Comment:
This may also require importing QUrl.

```suggestion
#include <QHBoxLayout>
#include <QDesktopServices>
```